custom-eslint-rules/catch-blocks/enforce-catch-block-error-type-guards.js

```diff
@@ -0,0 +1,86 @@
+module.exports = {
+	meta: {
+		type: 'suggestion',
+		version: '0.0.1',
+		name: 'eslint-plugin-enforce-catch-block-error-type-guards',
+		hasSuggestions: true,
+		fixable: 'code',
+		docs: {
+			description: 'Warn against the use of non type safe code in catch blocks',
+			category: 'Possible errors'
+		},
+		schema: []
+	},
+	create(context) {
+		const findParentNodeOfType = (node, parentType) => {
+			let parent = node;
+			while (parent) {
+				parent = parent.parent;
+				if (parent && parent.type === parentType) {
+					return parent;
+				}
+			}
+		};
+		return {
+			ExpressionStatement(node) {
+				const catchClause = findParentNodeOfType(node, 'CatchClause');
+				if (catchClause) {
+					const { param } = catchClause;
+					const value = node.expression.right;
+					if (!value) return;
+					let isProblem = false;
+					let prop;
+					if (
+						value.type === 'MemberExpression' &&
+						value.object &&
+						value.object.name === param.name
+					) {
+						prop = value.property;
+						let e = new Error();
+						// eslint-disable-next-line no-prototype-builtins
+						if (!e.hasOwnProperty(prop.name)) {
+							let ifStatement = findParentNodeOfType(node, 'IfStatement');
+							if (!ifStatement) {
+								isProblem = true;
+							} else {
+								if (!ifStatement.test.left.name === param.name) isProblem = true;
```

Comment:
This appears to be where the error is, also it'd be clearer as `ifStatement.test.left.name !== param.name` if that is what is intended.